File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/05_While_Loop_Exercise/06_Cake.js
function cake(input) {
    let width = Number(input[0]);
    let length = Number(input[1]);

    let cakePieces = width * length;
    let leftPieces = cakePieces;


    let i = 2;
    while (true) {
        let currentCommand = input[i];
        i++;

        if (currentCommand == "STOP") {
            console.log(`${leftPieces} pieces are left.`);
            break;
        }

        let currentPieces = Number(currentCommand);
        leftPieces -= currentPieces;

        if (leftPieces < 0) {
            console.log(`No more cake left! You need ${-leftPieces} pieces more.`);  // Math.abs(leftPieces)
            break;
        }
    }
}


// cake(["10",
//     "10",
//     "20",
//     "20",
//     "20",
//     "20",
//     "21"]);

// cake(["10",
//     "2",
//     "2",
//     "4",
//     "6",
//     "STOP"]);

File: Python_Courses/Python_Basics/Preparation/05_While_Loop_Exercise/06_Cake.py
STOP_COMMAND = "STOP"

cake_width = int(input())
cake_length = int(input())

cake_pieces = cake_width * cake_length

while True:
    command = input()
    if command == STOP_COMMAND:
        break

    current_pieces = int(command)
    cake_pieces -= current_pieces
    if cake_pieces <= 0:
        break

if cake_pieces > 0:
    print(f"{cake_pieces} pieces are left.")
else:
    print(f"No more cake left! You need {-cake_pieces} pieces more.")
